controllers/api/file/PKPManageFileApiHandler.inc.php

```diff
@@ -171,11 +180,20 @@ function saveMetadata($args, $request) {
 			$metadataForm->execute($args, $request);
 			$submissionFile = $metadataForm->getSubmissionFile();
 
+			// Get a list of author user IDs
+			$authorUserIds = array();
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+			$submitterAssignments = $stageAssignmentDao->getBySubmissionAndRoleId($submission->getId(), ROLE_ID_AUTHOR);
+			while ($assignment = $submitterAssignments->next()) {
+				$authorUserIds[] = $assignment->getUserId();
+			}
```

Comment:
I thought about that, but it's only 3 cases, and I don't like when DAO methods mix a lot of concerns together about different tables/objects.